pkg/build/stage/git_mapping.go

```diff
@@ -108,6 +109,31 @@ func (gm *GitMapping) getPathMatcher() path_matcher.PathMatcher {
 	})
 }
 
+// Generate a map with files to rename.
+// Returns a map with one element if `git[].add` is a file and its base filename differs from `git[].to` base filename,
+// otherwise return an empty map. The key is full path to `git[].add` file and the value is base filename of `git[].to` file.
+func (gm *GitMapping) getFileRenames(ctx context.Context) (map[string]string, error) {
```

Comment:
Change to something like getArchiveOptions() and add at least PathScope generation?